teps/0101-decouple-catalog-organization-and-reference.md

```diff
@@ -0,0 +1,233 @@
+- [Proposal](#proposal)
+  - [Notes/Caveats (optional)](#notescaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [User Experience (optional)](#user-experience-optional)
+  - [Performance (optional)](#performance-optional)
+- [Design Details](#design-details)
+- [Test Plan](#test-plan)
+- [Design Evaluation](#design-evaluation)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [Upgrade &amp; Migration Strategy (optional)](#upgrade--migration-strategy-optional)
+- [Implementation Pull request(s)](#implementation-pull-request-s)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+As described in [TEP-0003](./0003-tekton-catalog-organization.md) :
+The Tekton catalog is a collection of blessed Tekton resources that
+can be used with any system that supports the Tekton API. We also have
+the [hub](https://hub.tekton.dev) to be able to search for Tasks.
+
+This proposes to make the *Hub* and any other **official** mean of
+consuming Tekton resources from the community decoupled from where we
+author them. In a gist, this aims to make the repository
+`tektoncd/catalog` an implementation detail, and only support
+long-term, non-changing URI to refer to tasks (Those URIs could be
+`https://` endpoint such as the Hub or docker images, …).
+
+## Motivation
+
+As of today, users of the tekton task are directly refering to the
+GitHub URL to get Task from the catalog. This creates a hard coupling
+between the way we organize tasks inside Tekton and the way user
+consume it. This is forcing us into one possible catalog organization
+and prevent us to make drastic changes to this organization without
+breaking users.
+
+This enhancement aim to decouple the catalog organization from users
+consumption of those resources. This enhancement proposal is mainly
+touching the catalog and the hub components. This would also push the
+hub to be the main entrypoint for users to search, list and get tekton
+resources.
+
+### Goals
+
+- Allowing for the maintainers of `tektoncd/catalog` to change the
+  organization without affecting the consumption of the resources.
+- Define a set of standard, supported ways to get a tekton resource
+  (`Task`, `Pipeline`, …)
+
+### Non-Goals
+
+- Modify the current organization of the `tektoncd/catalog` repository.
+
+### Use Cases
+
+<!--
+Describe the concrete improvement specific groups of users will see if the
+Motivations in this doc result in a fix or feature.
+
+Consider both the user's role (are they a Task author? Catalog Task user?
+Cluster Admin? etc...) and experience (what workflows or actions are enhanced
+if this problem is solved?).
+-->
+
+## Requirements
+
+- A end-user should never refer / install task from a URL that points
+  directly to the github repository (like https://raw.githubusercontent.com/tektoncd/catalog/main/task/git-clone/0.5/git-clone.yaml).
```

Comment:
Is this in reference to that specific example of using the raw path, or referring to git repos in general? wondering because the git resolver proposed in [TEP-0060: Remote Resolution](https://github.com/tektoncd/community/blob/main/teps/0060-remote-resource-resolution.md#git-resolver) expects to use a specific path for a given resource - just wanted to clarify what this means with respect to that, I was under the impression that we wanted